lms/djangoapps/grades/signals/handlers.py

```diff
@@ -144,6 +145,8 @@ def score_published_handler(sender, block, user, raw_earned, raw_possible, only_
             modified=score_modified_time,
             score_db_table=ScoreDatabaseTableEnum.courseware_student_module,
         )
+        if isinstance(block, ScorableXBlockMixin):
```

Comment:
ducktyping is generally preferable - that is, checking for `getattr(block, 'set_score')` instead of checking for a type.